src/lib.rs

```diff
@@ -376,20 +399,32 @@ fn compute_initial_info<'a, D: BidiDataSource, T: TextSource<'a> + ?Sized>(
             _ => {}
         }
     }
-    if para_start < text.len() {
-        paragraphs.push(ParagraphInfo {
-            range: para_start..text.len(),
-            level: para_level.unwrap_or(LTR_LEVEL),
-        });
-        pure_ltr.push(is_pure_ltr);
+
+    if split_paragraphs {
+        if para_start < text.len() {
+            paragraphs.as_mut().unwrap().push(ParagraphInfo {
+                range: para_start..text.len(),
+                level: para_level.unwrap_or(LTR_LEVEL),
+            });
+            pure_ltr.as_mut().unwrap().push(is_pure_ltr);
+        }
+        assert_eq!(
```

Comment:
debug assertion please

Reply:
OK, will do (although this was essentially just moving the [already-existing assertion](https://github.com/servo/unicode-bidi/blob/master/src/lib.rs#L387) inside the `if split_paragraphs` condition).